### ISSUE TITLE
Mismatched types false positives

### ISSUE THREAD

**Issue by Teln0** [intellij-rust/intellij-rust#5564]
## Environment

* **IntelliJ Rust plugin version:** No idea where to find that, but probably latest, I installed it recently.
* **Rust toolchain version:** cargo 1.43.0
* **IDE name and version:** Clion (2020.1.2) and Intelij (2020.1)
* **Operating system:** Ubuntu linux

## Problem description
The issue occurs in both clion and intelij.
It gives me a false positive for mismatched types (expected &String found &str)

## Steps to reproduce
```rust
fn main() {
    let a: String = "test".to_string();
    if !a.eq("test2") {
        println!("OK");
    }
}
```

It tells me "mismatched types" on "test2" (expected &String found &str), when the compiler doesn't complain at all and everything works fine. But if I change "test2" to "test2".to_string().borrow(), the IDE (both intelij and clion) doesn't complain but now the compiler throws an error.

I've also created an issue [here](https://youtrack.jetbrains.com/issue/CPP-20773)

**Comment by Undin** [comment]
> No idea where to find that, but probably latest, I installed it recently.

<img width="999" alt="Screen Shot 2020-06-13 at 20 45 25" src="https://user-images.githubusercontent.com/2539310/84575604-eec89f80-adb6-11ea-9e92-f88e9c941004.png">

Also, there is helpful action `Help | Create New Issue` to create a template of new issue in our repo just from your IDE. It fills environment fields automatically. Also, if you have selected code in your current editor, it adds it into `Steps to reproduce` section as code example

**Comment by Undin** [comment]
Try to enable new experimental macro expansion engine - it should fix this issue. See #3628 for more details

**Comment by Undin** [comment]
Duplicate of #4224

**Comment by Teln0** [comment]
Fixed it for me, thanks
